tests/Auth/AuthTokenGuardTest.php

```diff
@@ -171,6 +171,16 @@ public function testUserCanBeRetrievedByAuthHeadersWithCustomField()
         $this->assertEquals(1, $user->id);
     }
 
+    public function testUserCanBeRetrievedByBearerTokenWithCustomHeader()
+    {
+        $provider = m::mock(UserProvider::class);
+        $provider->shouldReceive('retrieveByCredentials')->once()->with(['api_token' => 'foo'])->andReturn((object) ['id' => 1]);
+        $request = Request::create('/', 'GET', [], [], [], ['HTTP_CUSTOM_BEARER' => 'Bearer foo']);
+        $guard = new TokenGuard($provider, $request, 'api_token', 'api_token', false, 'CUSTOM_BEARER');
```

Comment:
Your keys are different here and the test still passes so there's something awkward going on. Might I also suggest to use an actual example like you did in your PR comment and use the `X-Custom-Auth` name. That way it's much more clear why this would be needed.

Reply:
For the test case, the header names are case-insensitive (according to RFC 2616) which is why it passes. The header value is case sensitive, however, so the test should fail if the token had different capitalization.

Reply:
Nvm, I read that wrong :)

Reply:
Oh and for the different keys: The $_SERVER super global appends HTTP_ to the http request header...I verified that different headers cause the test case to fail..